hnn_core/network_builder.py

```diff
@@ -443,40 +443,43 @@ def _connect_celltypes(self, src_type, target_type, loc,
             If True, allow connecting neuron to itself.
         """
         net = self.net
-        connection_name = f'{src_type}_{target_type}_{receptor}'
-        if connection_name not in self.ncs:
-            self.ncs[connection_name] = list()
+
         assert len(self.cells) == len(self._gid_list) - len(self._drive_cells)
-        # NB this assumes that REAL cells are first in the _gid_list
-        for gid_target, target_cell in zip(self._gid_list, self.cells):
-            is_target_gid = (gid_target in
-                             self.net.gid_ranges[_long_name(target_type)])
-            if _PC.gid_exists(gid_target) and is_target_gid:
-                gid_srcs = net.gid_ranges[_long_name(src_type)]
-                if unique:
-                    gid_srcs = [gid_target + net.gid_ranges[src_type][0]]
-                for gid_src in gid_srcs:
-
-                    if not allow_autapses and gid_src == gid_target:
-                        continue
-
-                    pos_idx = gid_src - net.gid_ranges[_long_name(src_type)][0]
-                    # NB pos_dict for this drive must include ALL cell types!
-                    nc_dict['pos_src'] = net.pos_dict[
-                        _long_name(src_type)][pos_idx]
-
-                    # get synapse locations
-                    syn_keys = list()
-                    if loc in ['proximal', 'distal']:
-                        for sect in target_cell.sect_loc[loc]:
-                            syn_keys.append(f'{sect}_{receptor}')
-                    else:
-                        syn_keys = [f'{loc}_{receptor}']
-
-                    for syn_key in syn_keys:
-                        nc = target_cell.parconnect_from_src(
-                            gid_src, nc_dict, target_cell.synapses[syn_key])
-                        self.ncs[connection_name].append(nc)
+        # Gather indeces of targets on current node
+        connectivity = [conn for conn in connectivity if
+                        _PC.gid_exists(conn['target_gid'])]
+        target_gids = [conn['target_gid'] for conn in connectivity]
+        target_filter = {}
+        for idx in range(len(self.cells)):
+            gid = self._gid_list[idx]
+            if gid in target_gids:
+                target_filter[gid] = idx
+
+        for conn in connectivity:
+            src_type, src_gid = conn['src_type'], conn['src_gid']
+            target_type, target_gid = conn['target_type'], conn['target_gid']
+            target_cell = self.cells[target_filter[target_gid]]
+            connection_name = f'{_short_name(src_type)}_'\
```

Comment:
Is there a strong reason we keep on using `_short_name`? It's hard to keep track of which you're supposed to use when... I tried to consistently just use the long forms ('L2_pyramidal') etc. in the new drives-API.

Reply:
The param files and class names are both short names. I think we should consistently change to the short version in a separate PR (see #148). If we change to the long version, we'll have to also change the param file format (harder)